server/service/src/stocktake/update/generate.rs

```diff
@@ -114,6 +114,7 @@ fn generate_stock_in_out_or_update(
             cost_price_per_pack,
             sell_price_per_pack,
             expiry_date,
+            item_variant_id: stocktake_line.line.item_variant_id.clone(),
```

Comment:
Should this not follow the same pattern as the other fields?

```rust
let item_variant_id = row.item_variant_id.or(stock_line_row.item_variant_id)
```

Though it appears to be retaining the existing stock line data fine, because the initial stocktake line is created based on the stock line data in the frontend - but I think backend should continue to support partial updates

Reply:
I think it we do that we wouldn't be able to clear the `item_variant_id` with a stock take.
Maybe that's a good trade off? 

I'm personally not sure how partial updates should/could work with stock take?
If we leave `stocktake_line` columns null for things we don't want to change, how do we know we've captured all the data?


@andreievg thoughts?

I've updated for now, to just add a comment explaining...

Reply:
Added this, which I think alleviates the concern that item_variant it might not be populated in stocktake_line. https://github.com/msupply-foundation/open-msupply/pull/5552/commits/6bf227193ace8efa5baea5dd23e0cd1a2a01e7bc

I think I'll go ahead and merge, but please feel free to create another issue if we think further cleanup is required...